android/java/org/chromium/chrome/browser/crypto_wallet/adapters/TwoLineItemRecyclerViewAdapter.java

```diff
@@ -36,44 +50,76 @@ public TwoLineItemRecyclerViewAdapter(List<TwoLineItem> items) {
     public RecyclerView.ViewHolder onCreateViewHolder(ViewGroup parent, int viewType) {
         if (viewType == TwoLineItem.TYPE_DIVIDER) {
             return new ViewHolderDivider(
-                    LayoutInflater.from(parent.getContext())
+                    getInflater(parent.getContext())
                             .inflate(R.layout.item_fragment_two_line_divider_item, parent, false));
+        } else if (viewType == TwoLineItem.TYPE_SINGLE) {
+            return new ViewHolderSingleText(
+                    getInflater(parent.getContext())
+                            .inflate(R.layout.item_fragment_two_line_single_text, parent, false));
         }
         int layout = R.layout.item_fragment_two_line_item;
         if (ORIENTATION.HORIZONTAL == mOrientation) {
             layout = R.layout.item_two_line_horizontal;
         }
-        return new ViewHolder(
-                LayoutInflater.from(parent.getContext()).inflate(layout, parent, false));
+        return new ViewHolder(getInflater(parent.getContext()).inflate(layout, parent, false));
+    }
+
+    private LayoutInflater getInflater(Context context) {
+        if (mLayoutInflater == null) {
+            mLayoutInflater = LayoutInflater.from(context);
+        }
+        return mLayoutInflater;
     }
 
     @Override
     public void onBindViewHolder(final RecyclerView.ViewHolder holder, int position) {
         TwoLineItem twoLineItem = mValues.get(position);
         if (twoLineItem.getType() == TwoLineItem.TYPE_TEXT) {
             ViewHolder viewHolder = (ViewHolder) holder;
+            // Only vertical layout support image icon and blockies
+            if (ORIENTATION.VERTICAL == mOrientation) {
```

Comment:
I think that wallet orientation is locked to vertical anyways.

Reply:
It's orientation for adapter items in list not related to activity or fragment specific.

Reply:
Renamed for more clarity https://github.com/brave/brave-core/pull/20168/commits/585cb2d0aae73475ccd85aefa08569497ad3ae03